core/src/main/java/io/keen/client/java/KeenProject.java

```diff
@@ -28,19 +28,35 @@ public KeenProject() {
      *                  for writing events.
      */
     public KeenProject(String projectId, String writeKey, String readKey) {
-        if (projectId == null || projectId.length() == 0) {
+        this(projectId, writeKey, readKey, null);
+    }
+
+    /**
+     * Construct a Keen project.
+     *
+     * @param projectId The Keen IO Project ID.
+     * @param writeKey  Your Keen IO Write Key. This may be null if this project will only be used
+     *                  for reading events.
+     * @param readKey   Your Keen IO Read Key. This may be null if this project will only be used
+     *                  for writing events.
+     * @param masterKey Your Keen IO Master Key. This may be null if not using administrative
+     *                  functionality that requires such a key.
+     */
+    public KeenProject(String projectId, String writeKey, String readKey, String masterKey) {
+        if (projectId == null || projectId.trim().isEmpty()) {
```

Comment:
What sort of validation? Since this is tossed about between modules, we can't yet know if a master is needed or a write or a read, because you may only be doing queries, or only doing administrative operations, or only doing writes, or whatever, so in theory we can't know what's needed. That said, at least one of them should be non-null and non-empty, so at a minimum we could validate that.